zScanner/SupportFiles/Extensions/UITextField.swift

```diff
@@ -9,47 +9,17 @@
 import UIKit
 
 extension UITextField {
-    func setBottomBorder(show: Bool = true, color: CGColor = UIColor.white.cgColor, animated: Bool = false, duration: CFTimeInterval = 1) {
+    func setBottomBorder(show: Bool = true, animated: Bool = true, color: UIColor = .primary, duration: CFTimeInterval = 0.333) {
         borderStyle = .none
         
         layer.backgroundColor = UIColor.white.cgColor
         layer.masksToBounds = false
-        layer.shadowColor = show ? color : UIColor.clear.cgColor
-        layer.shadowOffset = CGSize(width: 0.0, height: 1.0)
+        layer.shadowColor = show ? color.cgColor : UIColor.clear.cgColor
         layer.shadowRadius = 0.0
+        self.layer.shadowOffset = CGSize(width: 0, height: 1)
         
-        if animated {
-            let animation:CABasicAnimation = CABasicAnimation(keyPath: "shadowOpacity")
-            animation.fromValue = 0
-            animation.toValue = 1
-            animation.duration = duration
-            layer.add(animation, forKey: "Opacity")
-            layer.shadowOpacity = 1
-        }
```

Comment:
This seems too complicated code for a simple animation, usually is enough if you wrap the change you want to animate in
`UIView.animate()` function

Reply:
Wow, that's much easier, thank you.